dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -259,6 +278,31 @@ private void logRetry(long backoffNanoseconds, @Nullable Throwable throwable) {
         }
     }
 
+    private static boolean safeAndIdempotent(HttpMethod httpMethod) {
+        switch (httpMethod) {
+            case GET:
+            case HEAD:
+                // As per https://tools.ietf.org/html/rfc7231#section-4.2.1. "Safe" means roughly read-only.
+                // "Of the request methods defined by this specification, the GET, HEAD, OPTIONS, and TRACE methods
+                // are defined to be safe."
+                return true;
+            case PUT:
+            case DELETE:
```

Comment:
Initial plan is to not do PUT/DELETE because we're confident that devs have made them idempotent. Can revisit when we've done some more thorough investigation (e.g. analyzing request logs)